azure-pipelines.yml

```diff
@@ -0,0 +1,70 @@
+# Python package
+# Create and test a Python package on multiple Python versions.
+# Add steps that analyze code, save the dist with the build record, publish to a PyPI-compatible index, and more:
+# https://docs.microsoft.com/azure/devops/pipelines/languages/python
+
+trigger:
+- master
+
+strategy:
+  maxParallel: 3
+  matrix:
+    Ubuntu-Py36:
+      imageName: 'ubuntu-latest'
+      python.version: '3.6'
+    Ubuntu-Py37:
+      imageName: 'ubuntu-latest'
+      python.version: '3.7'
+    Windows:
+      imageName: 'vs2017-win2016'
+      python.version: '3.7'
+    MacOS:
+      imageName: 'macos-10.14'
+      python.version: '3.7'
+
+
+pool:
+  vmImage: '$(imageName)'
+
+steps:
+- task: UsePythonVersion@0
+  inputs:
+    versionSpec: '$(python.version)'
+  displayName: 'Use Python $(python.version)'
+
+- script: |
+    python -m pip install --upgrade pip
+    python -m pip install colcon-core colcon-output distro
+
+  displayName: 'Install dependencies'
+
+- script: |
```

Comment:
I think that the failing windows job here
https://dev.azure.com/osrf/colcon-sanitizer-reports/_build/results?buildId=9
is actually due to it not understanding all these backslashes. We might want to chance this to
`python -m pip install -r requirements.txt`

Reply:
Yeah that's definitely part of it. I am out of the loop on the current best practices for Python dependencies as I would have expected test and install dependencies to be centralized into setup.cfg not copied back out again.

We're also going to run into trouble if future scripts use bourne shellisms rather than single command invocations. More and more I think we should just ditch Windows CI for now.

Reply:
as I scroll I see you've already added a requirements.txt so don't mind me.